qa/tasks/cephfs/test_nfs.py

```diff
@@ -141,15 +141,31 @@ def _test_idempotency(self, cmd_func, cmd_args):
         if ret != 0:
             self.fail("Idempotency test failed")
 
-    def _test_create_cluster(self):
+    def _test_create_cluster(self, _sleep=3, _tries=3):
         '''
         Test single nfs cluster deployment.
+        :param _sleep: The amount of time to sleep between tries (default 3).
+        :param _tries: The amount of tries before giving up (default 3).
         '''
-        # Disable any running nfs ganesha daemon
-        self._check_nfs_server_status()
-        self._nfs_cmd('cluster', 'create', self.cluster_id)
-        # Check for expected status and daemon name (nfs.<cluster_id>)
-        self._check_nfs_cluster_status('running', 'NFS Ganesha cluster deployment failed')
+        with contextutil.safe_while(sleep=_sleep, tries=_tries) as proceed:
+            while proceed():
+                try:
+                    # Disable any running nfs ganesha daemon
+                    self._check_nfs_server_status()
+                    self._nfs_cmd('cluster', 'create', self.cluster_id)
+                    # Check for expected status and daemon name
+                    # (nfs.<cluster_id>)
+                    self._check_nfs_cluster_status(
+                        'running', 'NFS Ganesha cluster deployment failed')
+                    break
+                except (AssertionError, CommandFailedError) as e:
+                    log.warning(f'{e}, retrying')
+                    last_error = str(e)
```

Comment:
Is the `last_error` useful?

Reply:
can be helpful to identify what caused the failure while debugging

Reply:
else it would just say something like `MaxWhileTries: reached maximum tries (X) after waiting for Y seconds`

Reply:
> Is the `last_error` useful?
> 
> EDIT: since its gets logged anyway.

Yeah it would log at `log.warning()` but you would need to find that string, with `log.error('error encountered before timing out: 'f'{last_error}')`, it would be easy to see the actual error along with `MaxWhileTries: reached maximum tries (X) after waiting for Y seconds` error. if you think its too much verbose then i dont mind dropping this

Reply:
We already log, so its fine to drop this.

Reply:
sure

Reply:
dropped